starter-kits/nextjs-starter-kit/components/NoteEditor/NoteHeader.tsx

```diff
@@ -0,0 +1,156 @@
+import { BlockNoteEditor } from "@blocknote/core";
+import { useMutation, useSelf, useStorage } from "@liveblocks/react";
+import { Icon } from "@liveblocks/react-ui";
+import { useState } from "react";
+import { Title } from "@/components/NoteEditor/Title";
+import { CrossIcon, ImageIcon } from "@/icons";
+import { Button } from "@/primitives/Button";
+import { MyEmojiPicker } from "@/primitives/EmojiPicker";
+import { Popover } from "@/primitives/Popover";
+import styles from "./NoteHeader.module.css";
+
+export function NoteHeader({ editor }: { editor: BlockNoteEditor | null }) {
+  return (
+    <div className={styles.noteHeader}>
+      <Cover />
+      <EmojiAndButtons />
+      <Title editor={editor} />
+    </div>
+  );
+}
+
+// Use Liveblocks Storage to set/read real-time properties for cover colours and emoji icons
+function EmojiAndButtons() {
+  const cover = useStorage((root) => root.cover);
+  const icon = useStorage((root) => root.icon);
+  const canWrite = useSelf()?.canWrite;
+
+  const [coverPopoverOpen, setCoverPopoverOpen] = useState(false);
+  const [emojiPopoverOpen, setEmojiPopoverOpen] = useState(false);
+  const isPopoverOpen = coverPopoverOpen || emojiPopoverOpen;
+
+  const setCover = useMutation(({ storage }, newColor: string | null) => {
+    storage.set("cover", newColor);
+  }, []);
+
+  const setIcon = useMutation(({ storage }, newIcon: string | null) => {
+    storage.set("icon", newIcon);
+  }, []);
+
+  if (!canWrite) {
+    return (
+      <div className={styles.buttonRow}>
+        {icon ? <div className={styles.readOnlyIcon}>{icon}</div> : null}
+      </div>
+    );
+  }
+
+  return (
+    <div
+      className={styles.buttonRow}
+      data-icon={icon !== null || undefined}
+      data-popover-open={isPopoverOpen || undefined}
+    >
+      <Popover
+        open={emojiPopoverOpen}
+        onOpenChange={setEmojiPopoverOpen}
+        content={
+          <MyEmojiPicker
+            onEmojiSelect={(emoji) => {
+              setIcon(emoji);
+              setEmojiPopoverOpen(false);
+            }}
+            buttonSlot={
+              <Button
+                variant="subtle"
+                onClick={() => {
+                  setIcon(null);
+                  setEmojiPopoverOpen(false);
+                }}
+              >
+                Remove
+              </Button>
```

Comment:
Could this be a cross like the colors one? Also, with the label and the search input's fixed width, there's an empty space in the list instead of fitting.

![picker-width](https://github.com/user-attachments/assets/491e1ea4-6cc4-46cf-b5b6-602496f73ec9)